File: Android /Android/25:Jan:2017/FishbowlMobileApps/AndroidSDK/FishbowlMobileApps-Jamba-Android-v2.4.1/JMBA_InComm_Android_SDK/app/src/main/java/com/wearehathway/apps/incomm/Models/InCommSubmitPayment.java
package com.wearehathway.apps.incomm.Models;

/**
 * Created by Nauman Afzaal on 06/08/15.
 */
public class InCommSubmitPayment
{
    private double Amount;
    private String City;
    private String Country;
    private int CreditCardExpirationMonth;
    private int CreditCardExpirationYear;
    private String CreditCardNumber;
    private String CreditCardVerificationCode;
    private String FirstName;
    private String LastName;
    private int PaymentAccountId; //The optional UserPaymentAccount to use for paying for an Order if a User has stored their credit card info with their User account. Use this to create a new Order charging their payment account. The SubmitOrder request must be authenticated using the User credentials of the associated User.
    private boolean PaymentReceived;//If true, marks this order as paid at time of submission. No payment is taken and approval is not required.
    private String StateProvince;
    private String StreetAddress1;
    private String StreetAddress2;
    private String VestaOrgId;
    private String VestaWebSessionId;
    private String ZipPostalCode;
    private InCommOrderPaymentMobileDevice MobileDevice;
    private String CreditCardType;
    private String OrderPaymentMethod;

    public String getStreetAddress1()
    {
        return StreetAddress1;
    }

    public void setStreetAddress1(String streetAddress1)
    {
        StreetAddress1 = streetAddress1;
    }

    public String getStreetAddress2()
    {
        return StreetAddress2;
    }

    public void setStreetAddress2(String streetAddress2)
    {
        StreetAddress2 = streetAddress2;
    }

    public String getZipPostalCode()
    {
        return ZipPostalCode;
    }

    public void setZipPostalCode(String zipPostalCode)
    {
        ZipPostalCode = zipPostalCode;
    }

    public double getAmount()
    {
        return Amount;
    }

    public void setAmount(double amount)
    {
        Amount = amount;
    }

    public String getCity()
    {
        return City;
    }

    public void setCity(String city)
    {
        City = city;
    }

    public String getCountry()
    {
        return Country;
    }

    public void setCountry(String country)
    {
        Country = country;
    }

    public int getCreditCardExpirationMonth()
    {
        return CreditCardExpirationMonth;
    }

    public void setCreditCardExpirationMonth(int creditCardExpirationMonth)
    {
        CreditCardExpirationMonth = creditCardExpirationMonth;
    }

    public int getCreditCardExpirationYear()
    {
        return CreditCardExpirationYear;
    }

    public void setCreditCardExpirationYear(int creditCardExpirationYear)
    {
        CreditCardExpirationYear = creditCardExpirationYear;
    }

    public String getCreditCardNumber()
    {
        return CreditCardNumber;
    }

    public void setCreditCardNumber(String creditCardNumber)
    {
        CreditCardNumber = creditCardNumber;
    }

    public String getCreditCardVerificationCode()
    {
        return CreditCardVerificationCode;
    }

    public void setCreditCardVerificationCode(String creditCardVerificationCode)
    {
        CreditCardVerificationCode = creditCardVerificationCode;
    }

    public String getFirstName()
    {
        return FirstName;
    }

    public void setFirstName(String firstName)
    {
        FirstName = firstName;
    }

    public String getLastName()
    {
        return LastName;
    }

    public void setLastName(String lastName)
    {
        LastName = lastName;
    }

    public int getPaymentAccountId()
    {
        return PaymentAccountId;
    }

    public void setPaymentAccountId(int paymentAccountId)
    {
        PaymentAccountId = paymentAccountId;
    }

    public boolean isPaymentReceived()
    {
        return PaymentReceived;
    }

    public void setPaymentReceived(boolean paymentReceived)
    {
        PaymentReceived = paymentReceived;
    }

    public String getStateProvince()
    {
        return StateProvince;
    }

    public void setStateProvince(String stateProvince)
    {
        StateProvince = stateProvince;
    }

    public String getVestaOrgId()
    {
        return VestaOrgId;
    }

    public void setVestaOrgId(String vestaOrgId)
    {
        VestaOrgId = vestaOrgId;
    }

    public String getVestaWebSessionId()
    {
        return VestaWebSessionId;
    }

    public void setVestaWebSessionId(String vestaWebSessionId)
    {
        VestaWebSessionId = vestaWebSessionId;
    }

    public InCommOrderPaymentMobileDevice getMobileDevice()
    {
        return MobileDevice;
    }

    public void setMobileDevice(InCommOrderPaymentMobileDevice mobileDevice)
    {
        MobileDevice = mobileDevice;
    }

    public String getCreditCardType()
    {
        return CreditCardType;
    }

    public void setCreditCardType(String creditCardType)
    {
        CreditCardType = creditCardType;
    }

    public String getOrderPaymentMethod()
    {
        return OrderPaymentMethod;
    }

    public void setOrderPaymentMethod(String orderPaymentMethod)
    {
        OrderPaymentMethod = orderPaymentMethod;
    }
}

File: Android /Android/25:Jan:2017/FishbowlMobileApps/AndroidSDK/FishbowlMobileApps-jamba-development/JMBA_InComm_iOS_SDK/InCommSDK/Models/Request/InCommSubmitPayment.swift
//
//  InCommSubmitPayment.swift
//  InCommSDK
//
//  Created by Taha Samad on 8/18/15.
//  Copyright (c) 2015 Fishbowl. All rights reserved.
//

import Foundation
import SwiftyJSON

public struct InCommSubmitPayment {

    public var amount: Double
    public var city: String?
    public var country: String?
    public var creditCardExpirationMonth: UInt16?
    public var creditCardExpirationYear: UInt16?
    public var creditCardNumber: String?
    public var creditCardVerificationCode: String?
    public var firstName: String?
    public var lastName: String?
    public var paymentAccountId: Int32?
    public var paymentReceived: Bool?
    public var stateProvince: String?
    public var streetAddress1: String?
    public var streetAddress2: String?
    public var vestaOrgId: String?
    public var vestaWebSessionId: String?
    public var zipPostalCode: String?
    public var mobileDevice: InCommOrderPaymentMobileDevice?
    public var creditCardType: String?
    public var orderPaymentMethod: InCommOrderPaymentMethod?
    
    public init(amount: Double, orderPaymentMethod: InCommOrderPaymentMethod = .CreditCard, firstName: String? = nil, lastName: String? = nil, streetAddress1: String? = nil, streetAddress2: String? = nil, city: String? = nil, stateProvince: String? = nil, country: String? = nil, zipPostalCode: String? = nil, creditCardNumber: String? = nil, creditCardVerificationCode: String? = nil, creditCardExpirationMonth: UInt16? = nil, creditCardExpirationYear: UInt16? = nil, creditCardType: String?) {
        self.amount = amount
        self.orderPaymentMethod = orderPaymentMethod
        if orderPaymentMethod == .CreditCard {
            assert(firstName != nil, "firstName should be provided for payment method")
            assert(lastName != nil, "lastName should be provided for payment method")
            assert(streetAddress1 != nil, "streetAddress1 should be provided for payment method")
            assert(city != nil, "city should be provided for payment method")
            assert(stateProvince != nil, "stateProvince should be provided for payment method")
            assert(country != nil, "country should be provided for payment method")
            assert(zipPostalCode != nil, "zipPostalCode should be provided for payment method")
            assert(creditCardNumber != nil, "creditCardNumber should be provided for payment method")
            assert(creditCardVerificationCode != nil, "creditCardVerificationCode should be provided for payment method")
            assert(creditCardExpirationMonth != nil, "creditCardExpirationMonth should be provided for payment method")
            assert(creditCardExpirationYear != nil, "creditCardExpirationYear should be provided for payment method")
            assert(creditCardType != nil, "creditCardType should be provided for payment method")
        }
        self.firstName = firstName
        self.lastName = lastName
        self.streetAddress1 = streetAddress1
        self.streetAddress2 = streetAddress2
        self.city = city
        self.stateProvince = stateProvince
        self.country = country
        self.zipPostalCode = zipPostalCode
        self.creditCardNumber = creditCardNumber
        self.creditCardVerificationCode = creditCardVerificationCode
        self.creditCardExpirationMonth = creditCardExpirationMonth
        self.creditCardExpirationYear = creditCardExpirationYear
        self.creditCardType = creditCardType
    }
    
    public func serializeAsJSONDictionary() -> InCommJSONDictionary {
        var jsonDict                           = InCommJSONDictionary()
        jsonDict["Amount"]                     = amount
        jsonDict["City"]                       = city
        jsonDict["Country"]                    = country
        jsonDict["CreditCardExpirationMonth"]  = creditCardExpirationMonth != nil ? NSNumber(unsignedShort: creditCardExpirationMonth!) : nil
        jsonDict["CreditCardExpirationYear"]   = creditCardExpirationYear != nil ? NSNumber(unsignedShort: creditCardExpirationYear!) : nil
        jsonDict["CreditCardNumber"]           = creditCardNumber
        jsonDict["CreditCardVerificationCode"] = creditCardVerificationCode
        jsonDict["FirstName"]                  = firstName
        jsonDict["LastName"]                   = lastName
        jsonDict["PaymentAccountId"]           = paymentAccountId != nil ? NSNumber(int: paymentAccountId!) : nil
        jsonDict["PaymentReceived"]            = paymentReceived
        jsonDict["StateProvince"]              = stateProvince
        jsonDict["StreetAddress1"]             = streetAddress1
        jsonDict["StreetAddress2"]             = streetAddress2
        jsonDict["VestaOrgId"]                 = vestaOrgId
        jsonDict["VestaWebSessionId"]          = vestaWebSessionId
        jsonDict["ZipPostalCode"]              = zipPostalCode
        jsonDict["MobileDevice"]               = mobileDevice?.serializeAsJSONDictionary()
        jsonDict["CreditCardType"]             = creditCardType
        jsonDict["OrderPaymentMethod"]         = orderPaymentMethod?.rawValue
        return jsonDict
    }
}
